skyxplore-ui-test/src/test/java/com/mentoring/test/EquipmentTest.java

```diff
@@ -45,17 +78,34 @@ public void createAndSetUpAnEquipment() {
         assertThat(factoryPage.isExtenderItemInQueueVisible())
                 .as("Extender item in queue should be visible.")
                 .isTrue();
-        assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
+        softAssertion.assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
                 .as("Quantity in queue is not correct")
                 .isTrue();
-        assertThat(factoryPage.isQueueProcessStarted())
-                .as("Queue process is not started.")
+        softAssertion.assertThat(factoryPage.isQueueProcessBarVisible())
+                .as("Queue process bar is not visible.")
                 .isTrue();
 
-        //TODO: Subtask-3: Production time of CEX-01 is updated and item is finished
+        log.info("Updating queue process end time to finish the production.");
+        userId = DbSelects.getUserIdByEmailAddress(database, user.getEmail());
+        characterId = DbSelects.getCharacterIdByUserId(database, userId);
+        factoryId = DbSelects.getFactoryIdByCharacterId(database, characterId);
```

Comment:
I believe it would worth to move this to a separate method inside DB selects class. So if you'll need the factory ID multiple times, you wouldn't have to call all these three methods, but say for example: factoryId = DbSelects.getFactoryId(user.getEmail)

Reply:
I tried it. Please pull my new commit. 
1. There is an error :-( Can the runSelectStringValue handle 2 parameters in the SELECT?! Please discuss it.
2. Another question is here:  factoryId = DbSelects.getFactoryId(user.getEmail, characterName). In my solution I have to pass one more parameter: database. Please discuss it.

Reply:
You need to overload runSelectStringValue so it can handle 2 input values as well.

Reply:
Pls help with this.